### PR TITLE
Mgmt API updates

### PR DESCRIPTION
**Pull request by JoshLove-msft**
https://github.com/Azure/azure-sdk-for-net/issues/13353
https://github.com/Azure/azure-sdk-for-net/issues/13352

### CODE REVIEW

**Review by MiYanni — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Management/HttpRequestAndResponse.cs [hunk 1]
```diff
@@ -284,31 +276,16 @@ public async Task<string> GetTokenAsync(string requestUri)
...
-            }
+            await ThrowIfRequestFailed(request, response).ConfigureAwait(false);
+            return response;
         }
 
         private static async Task<string> ReadAsString(Response response)
         {
             string exceptionMessage;
-            using StreamReader reader = new StreamReader(response.ContentStream);
+            StreamReader reader = new StreamReader(response.ContentStream);
```
[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Management/HttpRequestAndResponse.cs:286]: woops I was supposed to change that back

**Review by schaabs — APPROVED** [review]

**Pull request merged**